In [1]:
import pandas as pd

from utils.config import load_config
from preprocessing.house_prices_preprocessing import HousePricesPreprocessor
from models.baseline.regression import run_baseline_regression

In [5]:
config = load_config()

data = pd.read_csv(config["paths"]["train"])

print("Data shape:", data.shape)

# --- разделяем target ---
target = "SalePrice"

y = data[target]
X = data.drop(columns=[target])

# --- preprocessing ---
preprocessor = HousePricesPreprocessor(outlier_quantile=0.95)

X_prepared = preprocessor.fit_transform(X)

X_prepared.head()

,MSSubClass,LotFrontage,LotArea,OverallQual,OverallCond,YearBuilt,YearRemodAdd,MasVnrArea,BsmtFinSF1,BsmtFinSF2,...,GarageFinish_Unf,GarageFinish_Fin,GarageFinish_None,BsmtFinType1_GLQ,BsmtFinType1_ALQ,BsmtFinType1_Unf,BsmtFinType1_Rec,BsmtFinType1_BLQ,BsmtFinType1_None,BsmtFinType1_LwQ
0,0.073375,-0.208804,-0.332210,0.651479,-0.517200,1.050994,0.878668,0.739648,0.667140,-0.327561,...,0,0,0,1,0,0,0,0,0,0
1,-0.872563,0.646406,-0.009592,-0.071836,2.179628,0.156734,-0.429577,-0.654947,1.327216,-0.327561,...,0,0,0,0,1,0,0,0,0,0
2,0.073375,-0.037762,0.453294,0.651479,-0.517200,0.984752,0.830215,0.497729,0.133255,-0.327561,...,0,0,0,1,0,0,0,0,0,0
3,0.309859,-0.493874,-0.023619,0.651479,-0.517200,-1.863632,-0.720298,-0.654947,-0.521967,-0.327561,...,1,0,0,0,1,0,0,0,0,0
4,0.073375,0.874462,1.297710,1.374795,-0.517200,0.951632,0.733308,1.835402,0.543376,-0.327561,...,0,0,0,1,0,0,0,0,0,0


In [6]:
results = run_baseline_regression(X_prepared, y)
results

,model,MAE,MSE,RMSE,R2
0,Ridge,19728.890159,1.094021e+09,33075.991003,0.857370
1,Lasso,20490.640935,1.157464e+09,34021.529266,0.849098
2,LinearRegression,20501.316464,1.158019e+09,34029.674064,0.849026
3,Dummy,59568.250000,7.861867e+09,88667.167506,-0.024972


# Baseline модели: результаты и выводы

## Задача

Предсказать цену дома (`SalePrice`) на основе признаков.

---

## Результаты baseline моделей

| Model            | MAE     | RMSE       | R²        |
| ---------------- |---------|------------| --------- |
| Ridge            | ~19 728 | **33 076** | **0.857** |
| Lasso            | ~20 490 | 34 021     | 0.849     |
| LinearRegression | ~20 501 | 34 029     | 0.849     |
| Dummy            | ~60 000 | 88 667     | ~0        |

---

## Интерпретация

### 1. Сравнение с Dummy

* Dummy модель (предсказывает медиану) даёт RMSE ≈ 87k
* Лучшая модель (Ridge) даёт RMSE ≈ 33k

Модель улучшает качество более чем в **2.5 раза**
Это означает, что модель действительно обучается и извлекает сигнал из данных

---

### 2. Ошибка модели (RMSE)

Средняя цена дома:

* **≈ 180 000**

Ошибка:

* **≈ 33 000**

[
Ошибка \approx 18%
]

Это **нормальный результат для baseline модели**

---

### 3. R² (коэффициент детерминации)

[
R^2 = 0.857
]

Модель объясняет **~85% вариации цены**

Интерпретация:

* 0 → ничего не объясняет
* 1 → идеальная модель

**0.85 — это сильный baseline**

---

## 🧩 Выводы

* Ridge показал лучший результат → регуляризация полезна
* Линейные модели уже дают хороший уровень качества
* Dummy сильно хуже → модель действительно обучается
* Ошибка находится в разумных пределах

---

# Планка качества (baseline threshold)

Устанавливаем baseline как ориентир:

## Основная метрика

* **RMSE ≈ 33 000**

## Целевая планка

* улучшение до **RMSE < 30 000**

